### PR TITLE
Add support for clicks on the chessboard

### PR DESCRIPTION
**Pull request by alexandrepiveteau**
This PR adds support for point-and-click to the `ChessBoard`. With point-and-click, each cell may be selected. Once a cell is selected, clicking a different cell will attempt to perform a move between the two cells, the nice side-effect being that clicking a cell twice will therefore unselect it (it attempts to perform the illegal move of moving a piece on itself). Whenever a piece is dragged and dropped, the current selection will disappear.

---

Here are some details over the changes of the PR :

- `ChessBoardState` now exposes a `val selectedPosition: ChessBoardState.Position`.
- `ChessBoardState.availableMoves` is now computed depending on `selectedPosition`.
- `ChessBoardState.onPositionClick(position: Position)` has been added. 
- A new `Modifier.cells()` has been added. It provides an easy API to draw in individual cells at some given positions of the board. It's used by a new `Modifier.selection()`, and `Modifier.actions()` has been refactored to use `Modifier.cells()` internally.
- The new behaviours are tested and the `ChessBoardRobot` has been slightly updated to provide support for more behaviours.

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoard.kt [hunk 1]
```diff
@@ -63,13 +62,29 @@ fun <Piece : ChessBoardState.Piece> ChessBoard(
...
     val cellPx = minDimension / ChessBoardCells
     val cellDp = with(LocalDensity.current) { cellPx.toDp() }
     val scope = rememberCoroutineScope()
 
+    // Detect the tap gestures on the board. This won't detect tap gestures on the pieces however.
+    Box(
+        Modifier.fillMaxSize().pointerInput(state, enabled) {
+          if (enabled) {
+            detectTapGestures { offset ->
+              val position = offset / cellPx
```
Comment (barmettlerl) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoard.kt:81]: Question: This divides x and y coordinates ?

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoard.kt [hunk 2]
```diff
@@ -143,33 +163,41 @@ private class DraggingState(
...
+              state.offset = currentTarget
+              state.isDragging = true
+            },
+            onDrag = { change, dragAmount ->
+              change.consumeAllChanges()
+              state.offset += dragAmount
+            },
+            onDragEnd = {
+              state.isDragging = false
+              val (x, y) = state.offset / cellSize
```
Comment (barmettlerl) [inline mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoard.kt:194]: Ah it divides directly both ^^

**Review by barmettlerl — APPROVED** [review]
Looks all good to me

File: mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoard.kt (hunk 2 above)
Comment (alexandrepiveteau) [reply mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoard.kt:194]: Yes. I discovered this when working on the first `ChessBoard` iteration, it's quite useful actually !

**Review by KurohanaJuri — APPROVED** [review]
LGTM

**Pull request merged**